ocaml/parsing/parsetree.mli

```diff
@@ -462,6 +462,8 @@ and expression_desc =
   | Pexp_extension of extension  (** [[%id]] *)
   | Pexp_unreachable  (** [.] *)
   | Pexp_stack of expression (** stack_ exp *)
+  | Pexp_overwrite of Longident.t loc * expression (** overwrite_ x with exp *)
```

Comment:
As discussed in person, I don't think the thing being overwritten needs to be a variable. But it's OK to go ahead with this for now and generalize later; leave a `(* CR uniqueness: ... *)` here to remind ourselves to return.